src/helm/benchmark/presentation/run_display.py

```diff
@@ -0,0 +1,267 @@
+
+    predictions = []
+    requests = []
+    for request_state in scenario_state.request_states:
+        assert request_state.instance.id is not None
+        if request_state.result is None:
+            continue
+
+        # For the multiple_choice_separate_calibrated adapter method,
+        # only keep the original prediction and discard the calibration prediction.
+        if (
+            run_spec.adapter_spec.method.startswith("multiple_choice_separate_calibrated")
+            and request_state.request_mode == "calibration"
+        ):
+            continue
+
+        stats_key = (
+            request_state.instance.id,
+            request_state.instance.perturbation,
+            request_state.train_trial_index,
+        )
+        trial_stats: Dict[str, float] = stats_by_trial[stats_key]
+        # For the multiple_choice_separate_* adapter methods,
+        # only keep the prediction for the chosen reference and discard the rest.
+        if (
+            run_spec.adapter_spec.method.startswith("multiple_choice_separate_")
+            and "predicted_index" in trial_stats
+            and trial_stats["predicted_index"] != request_state.reference_index
+        ):
+            continue
+
+        predicted_text = (
+            request_state.result.completions[0].text
+            if request_state.result is not None or request_state.result.completions
+            else None
+        )
+        mapped_output = (
+            request_state.output_mapping.get(predicted_text.strip()) if request_state.output_mapping else None
+        )
+
+        predictions.append(
+            DisplayPrediction(
+                instance_id=request_state.instance.id,
+                perturbation=request_state.instance.perturbation,
+                train_trial_index=request_state.train_trial_index,
+                predicted_text=predicted_text,
+                truncated_predicted_text=_truncate_predicted_text(predicted_text, request_state, run_spec.adapter_spec),
+                mapped_output=mapped_output,
+                reference_index=request_state.reference_index,
+                stats=trial_stats,
+            )
+        )
+        requests.append(
+            DisplayRequest(
+                instance_id=request_state.instance.id,
+                perturbation=request_state.instance.perturbation,
+                train_trial_index=request_state.train_trial_index,
+                request=request_state.request,
+            )
+        )
+
+    write(
+        os.path.join(run_path, "instances.json"),
+        json.dumps([asdict_without_nones(instance) for instance in scenario_state.instances], indent=2),
+    )
+    write(
+        os.path.join(run_path, "predictions.json"),
+        json.dumps(list(map(asdict_without_nones, predictions)), indent=2),
+    )
+    write(
+        os.path.join(run_path, "requests.json"),
+        json.dumps(list(map(asdict_without_nones, requests)), indent=2),
+    )
```

Comment:
Maybe the names should be `display_predictions.json` and `display_requests.json` so that it's a bit more obvious the function of these files...

Reply:
Changed; also changed the frontend to reflect this.